src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -46,7 +48,17 @@ public List<Book> findAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery("FROM Book", Book.class).getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't find all books", e);
+            throw new EntityNotFoundException("Can't find all books", e);
+        }
+    }
+
+    @Override
+    public Book findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> query = session.createQuery("FROM Book WHERE id = :id", Book.class);
+            return query.setParameter("id", id).uniqueResult();
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find book by id: " + id, e);
         }
     }
```

Comment:
Let's change this method to return an Optional<Book> and move the exception throwing to the service layer